framework/test/mocha/unit/components/storage/entities/transaction.js

```diff
@@ -263,42 +263,40 @@ describe('Transaction', () => {
 				id: transactions[0].id,
 			};
 			// Act & Assert
-			expect(() => {
+			await expect(() => {
 				transaction.get(validFilter);
 			}).not.to.throw(NonSupportedFilterTypeError);
 		});
 
 		it('should throw error for invalid filters', async () => {
 			const transaction = new Transaction(adapter);
-			try {
+
+			await expect(() => {
 				transaction.get({ invalid_filter: true });
-			} catch (err) {
-				expect(err.message).to.equal('One or more filters are not supported.');
-			}
+			}).to.throw('One or more filters are not supported.');
 		});
 
 		it('should accept only valid options', async () => {
 			const transaction = new Transaction(adapter);
-			return expect(
-				transaction.get({}, validOptions),
-			).to.eventually.fulfilled.and.deep.equal([]);
+			const res = await transaction.get({}, validOptions);
+
+			expect(res).to.deep.equal([]);
 		});
 
 		it('should throw error for invalid options', async () => {
 			const transaction = new Transaction(adapter);
-			try {
+
+			await expect(() => {
 				transaction.get({}, invalidOptions);
```

Comment:
this internal function is the `async`. I think it will be detached error if internal function throws

Reply:
I think it will not detach the error. If it would then this test case would have been failed.